### PR TITLE
Add login route to thimble

### PR DESCRIPTION
**Pull request by gideonthomas**
This will help with implementing the anon workflow

### CODE REVIEW

**Comment by humphd** [comment]
@sedge can you test and review this with @gideonthomas at CDOT, since I can't test.

File: public/scripts/sso-override.js [hunk 1]
```diff
@@ -4,70 +4,25 @@
...
 
-    setStateCookie(state);
+      window.location = loginUrl + (newUser ? "?signup=true" : "");
+    };
+  }
 
-    window.location = query;
-  });
+  // Signup login flow
+  joinEl.on('click', signIn(true));
```
Comment (sedge) [inline public/scripts/sso-override.js]: What happened to logout?

File: public/scripts/sso-override.js [hunk 2]
```diff
@@ -4,70 +4,25 @@
...
-      "&state=" + state,
-      "action=signin"
-    ].join("&");
-
-    setStateCookie(state);
-
-    window.location = query;
-  });
-
-  logoutEl.on('click', function(e) {
```
Comment (sedge) [inline public/scripts/sso-override.js]: We still need this

File: routes/config.js [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
 var Cryptr = require("cryptr");
 
+var env = require("../lib/environment");
+var oauth = env.get("OAUTH");
+var loginURL = oauth.authorization_url + "/login/oauth/authorize?" + [
+  "client_id=" + oauth.client_id,
+  "response_type=code",
+  "scopes=user email"
+].join("&");
+var logoutURL = oauth.authorization_url + "/logout?client_id=" + oauth.client_id;
```
Comment (sedge) [inline routes/config.js]: This should probably still live on the client.

File: views/userbar.html [hunk 4]
```diff
@@ -47,7 +47,7 @@
                   </a>
                 </li>
                 <li>
-                  <a href="#" id="logout-link" class="dropdown-link">
+                  <a href="{{ logoutURL }}" id="logout-link" class="dropdown-link" target="_self">
```
Comment (sedge) [inline views/userbar.html]: Ah.

**Comment by sedge** [comment]
Code looks good! Testing now...

**Comment by sedge** [comment]
r+

**Pull request merged**